### PR TITLE
Create healthcheck for cloud firestore

### PR DESCRIPTION
**Pull request by Odonno**
Linked to #146

### CODE REVIEW

**Comment by unaizorrilla** [comment]
Hi @Odonno 

I try to review it on next days, thanks for send this PR!!

File: AspNetCore.Diagnostics.HealthChecks.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28010.2036
+# Visual Studio Version 16
```
Comment (unaizorrilla) [inline AspNetCore.Diagnostics.HealthChecks.sln]: Can you mantain visual studio version?

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.Gcp.CloudFirestore
+{
+    public class CloudFirestoreHealthCheck : IHealthCheck
+    {
+        private readonly CloudFirestoreOptions _cloudFirestoreOptions = new CloudFirestoreOptions();
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: Remove new and ensure not null on parameter constructor with throw expression!

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        public CloudFirestoreHealthCheck(CloudFirestoreOptions cloudFirestoreOptions)
+        {
+            _cloudFirestoreOptions = cloudFirestoreOptions;
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: use cloudFirestoreOptions ?? throw new ArgumentNullException(nameof(cloudFirestoreOptions))

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var currentRootCollections = await GetRootCollectionsAsync(cancellationToken);
+
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: remove blank line

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
+        private async Task<List<string>> GetRootCollectionsAsync(CancellationToken cancellationToken)
+        {
+            var collections = new List<string>();
+            var enumerator = _cloudFirestoreOptions.FirestoreDatabase.ListRootCollectionsAsync().GetEnumerator();
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: ListRootCollectionAsync support cancellationToken?

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+            while (await enumerator.MoveNext(cancellationToken))
+            {
+                collections.Add(enumerator.Current.Id);
+            }
+
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: remove blank line

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                }
+
+                return HealthCheckResult.Healthy();
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: remove blank line

File: HealthChecks.Gcp.CloudFirestore/DependencyInjection/CloudFirestoreHealthCheckBuilderExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class CloudFirestoreHealthCheckBuilderExtensions
+    {
+        const string NAME = "cloud firestore";
+
+        public static IHealthChecksBuilder AddCloudFirestore(this IHealthChecksBuilder builder, Action<CloudFirestoreOptions> setup, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default)
```
Comment (unaizorrilla) [inline HealthChecks.Gcp.CloudFirestore/DependencyInjection/CloudFirestoreHealthCheckBuilderExtensions.cs]: Add missing comments on this public surface, please check existing comments on others projects.

File: HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs (hunk 5 above)
Comment (Odonno) [reply HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: No, there is only one method without any parameter.

Comment (unaizorrilla) [reply HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs]: ok!

**Comment by Odonno** [comment]
@unaizorrilla Done. Thanks for your help.

**Comment by unaizorrilla** [comment]
Hi @Odonno  

I try to merge this ASAP!

**Pull request merged**